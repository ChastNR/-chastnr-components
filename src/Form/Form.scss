.frm {
  display: flex;
  justify-content: center;
  margin: 2em auto;
  max-width: 450px;
  padding: 0 1em;

  & > div {
    display: flex;
    flex-direction: column;
    justify-content: center;
    width: 100%;

    .frm__title {
      margin-bottom: 2em;
      text-align: center;
    }
  }
}
