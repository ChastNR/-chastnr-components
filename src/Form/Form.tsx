import './Form.scss';

interface FormProps {
  title?: React.ReactNode;
}

const Form: React.FC<FormProps> = ({ children, title }) => (
  <div className="frm">
    <div>
      {title && (
        <div className="frm__title">
          <h2>{title}</h2>
        </div>
      )}
      {children}
    </div>
  </div>
);

export default Form;
