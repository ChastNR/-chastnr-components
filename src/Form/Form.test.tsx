import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Form from './Form';

describe('<Form />', () => {
  it('should equal snapshot', () => {
    const form = shallow(<Form title="Test Title">Test Content</Form>);
    expect(toJson(form)).toMatchSnapshot();
  });
});
