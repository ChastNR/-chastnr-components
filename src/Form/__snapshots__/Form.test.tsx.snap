// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should equal snapshot 1`] = `
<div
  className="frm"
>
  <div>
    <div
      className="frm__title"
    >
      <h2>
        Test Title
      </h2>
    </div>
    Test Content
  </div>
</div>
`;
